Modules/_winapi.c

```diff
@@ -150,6 +159,7 @@ overlapped_dealloc(OverlappedObject *self)
 
     CloseHandle(self->overlapped.hEvent);
     SetLastError(err);
+    PyObject_GC_UnTrack(self);
```

Comment:
I would prefer to *start* by untracking the GC, at the start to the dealloc function.

Reply:
I initially did that, but a concern was brought up here which I think makes sense https://github.com/python/cpython/pull/26381#discussion_r639921958.

Reply:
If tp_dealloc returns without clearing the object, IMO it's better to remove it from the GC. It's better to hide this badly broken Python object. I suggest to start by untracking and explain that it's a deliberate choice to untrack even if we hit the worst case scenario of "return" below.

Note: IMO this code should be removed, Windows XP is no longer supported, but it should be done in a separated change.

Reply:
Sorry, I just introduced a regression with this, seems like there's a race condition with the GC and the IO operation somewhere.

```
> python_d.exe -m test test_httplib -m test_local_bad_hostname 

...\cpython\Modules\gcmodule.c:446: update_refs: Assertion "gc_get_refs(gc) != 0" failed
Enable tracemalloc to get the memory block allocation traceback

object address  : 000002DADE2A99D0
object refcount : 0
object type     : 000002DADDF72F60
object type name: _winapi.Overlapped
object repr     : <refcnt 0 at 000002DADE2A99D0>

Fatal Python error: _PyObject_AssertFailed: _PyObject_AssertFailed
Python runtime state: initialized
```

It seems that placing the ``PyObject_GC_UnTrack`` at the top actually fixes it...